sapi/cli/php_cli.c

```diff
@@ -1208,6 +1212,11 @@ int main(int argc, char *argv[])
 	 */
 	argv = save_ps_args(argc, argv);
 
+#ifdef PHP_WIN32
```

Comment:
PHP_CLI_WIN32_NO_CONSOLE builds a binary without any interaction, which is meant to be run a a service for example. You can check sapi/cli/config.w32. Though i'm not sure how often this feature is actually used, I don't recall any related bug reports within last couple of years. Anyway  there is supposed to be no console interaction. Except maybe that the service is allowed to communicate with desktop, but even in that case - wouldn't the descriptors be redirected?

Thanks.